### PR TITLE
Implement ducktap autofunc

### PR DESCRIPTION
**Pull request by mxpph**
Adds a command, `+ducktap` which works as an automatic perfect duckroll. 
Also adds a cvar `cl_autojump_priority`, which controls whether ducktap or jump take precedence.

### CODE REVIEW

**Comment by YaLTeR** [comment]
> Also adds a cvar cl_ducktap_priority, which controls whether ducktap or jump take precedence.

Is this actually needed by somebody?

**Comment by mxpph** [comment]
> > Also adds a cvar cl_ducktap_priority, which controls whether ducktap or jump take precedence.
> 
> Is this actually needed by somebody?

Begi asked for it so I figured it would be a good idea

File: cl_dll/input.cpp [hunk 1]
```diff
@@ -778,7 +796,12 @@ void CL_DLLEXPORT CL_CreateMove ( float frametime, struct usercmd_s *cmd, int ac
 	//
 	cmd->buttons = CL_ButtonBits( 1 );
 
-	autofuncs::handle_autojump(cmd);
+	if (in_ducktap.state & 1)
+	{
+		cmd->buttons |= IN_DUCK;
+		autofuncs::handle_ducktap(cmd);
+	} else 
+		autofuncs::handle_autojump(cmd);
```
Comment (YaLTeR) [inline cl_dll/input.cpp:804]: Could you add a comment here saying that autojump overrides ducktap?

**Comment by YaLTeR** [comment]
Thanks!

**Pull request merged**